Makefile

```
@@ -30,7 +30,7 @@ generate:
 
 .PHONY: build
 build: generate
-	CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o otel-go-instrumentation cli/main.go
+	GOOS=linux GOARCH=$(uname -m) go build -o otel-go-instrumentation cli/main.go
```

looks like this change leaves `GOARCH` empty in the build CI (see https://github.com/open-telemetry/opentelemetry-go-instrumentation/actions/runs/4672079989/jobs/8273853096?pr=40#step:3:322):

```
GOOS=linux GOARCH= go build -o otel-go-instrumentation cli/main.go
```

but changing to this

```suggestion
	GOOS=linux GOARCH=$(shell uname -m) go build -o otel-go-instrumentation cli/main.go
```

gives me this build error:
```
GOOS=linux GOARCH=x86_64 go build -o otel-go-instrumentation cli/main.go
go: unsupported GOOS/GOARCH pair linux/x86_64
make: *** [Makefile:33: build] Error 2
```

---

Is there any opposition to adding this arch detection in a separate PR? I would like to not block this PR much longer and it's not directly related

---

Agree. Also, we don't support ARM yet (it is a separate PR) so building only for amd64, for now, makes sense.